yaml/shared/src/main/scala/org/virtuslab/yaml/YamlDecoder.scala

```diff
@@ -134,7 +134,7 @@ object YamlDecoder:
       override def construct(node: Node): Either[ConstructError, T] =
         node match
           case Node.MappingNode(mappings, _) =>
-            val valuesMap = mappings.map(e => e.key.value -> e.value).toMap
+            val valuesMap = mappings.map(e => e.key.asInstanceOf[ScalarNode].value -> e.value).toMap
```

Comment:
@kpodsiad What should happen if scalarNode doesn't appear there? 
throw an exception that such node isn't allowed?

Reply:
Yes, we should proper `ConstructError` because for product types key has to be an instance of ScalarNode